cloudsync/sync/state.py

```diff
@@ -1256,3 +1256,20 @@ def unconditionally_get_latest(self, ent, i):
             ent[i].path = info.path
             if ent.ignored == IgnoreReason.NONE and not ent[i].changed:
                 ent[i].changed = time.time()
+
+    def get_state_lookup(self, side: int) -> 'SyncStateLookup':
+        return SyncStateLookup(self, side)
+
+
+class SyncStateLookup:
+    """
+    Limited read-only SyncState interface
+    """
+
+    def __init__(self, state: 'SyncState', side: int):
```

Comment:
what's the rationale behind including `side` as a member variable instead of as a parameter in `get_path()`?

Reply:
Mainly that EventManager knows its side, but Provider (as of now) doesn't. Figured it is better to bake `side` into the SyncStateLookup instance than to add more state variables to the base Provider.

The plan is for Providers (likely only OneDrive) to do event filtering in the `events()` function instead of doing it in EventManager, and for that Providers need limited access to SyncState.

Reply:
this is great because gdrive maintains its own state now, and it can use this too instead

Reply:
box maintains state too... and all 3 of the providers do them in weirdly broken ways.

Reply:
okay cool, that sounds good to me. Approved